### PR TITLE
ci: Remove llvm 5, introduce llvm 16

### PR DESCRIPTION
**Pull request by emilio**
LLVM 5 is ancient.

### CODE REVIEW

**Comment by emilio** [comment]
r? @pvdrz (if you have the cycles)

Most of it is moving expectations around. I had to unfortunately remove the macOS llvm16 tests because the runner is on x86-64 but llvm provides only releases for arm macOS nowadays.

**Comment by emilio** [comment]
Okay I got macOS CI, but on 9.0 due to ^. Better than nothing tho.

File: CONTRIBUTING.md
Comment (kulp) [inline CONTRIBUTING.md:1]: Maybe it is time to remove [lines 228-235](https://github.com/rust-lang/rust-bindgen/pull/2719/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R228-R235) entirely?

(I forgot to do that in #2166.)

Comment (emilio) [reply CONTRIBUTING.md:1]: Yeah will do, thanks!

**Pull request merged**